app/main.py

```diff
@@ -1,8 +1,25 @@
 class Person:
-    # write your code here
-    pass
 
+    people = {}
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
+
+
+def create_person_list(people_list: list) -> list:
+    person_number = [
+        Person(person["name"], person["age"])
+        for person in people_list
+    ]
+
+    for person in people_list:
+        person_data = Person.people[person["name"]]
+
+        if "wife" in person and person["wife"] is not None:
+            person_data.wife = Person.people[person["wife"]]
+        elif "husband" in person and person["husband"] is not None:
+            person_data.husband = Person.people[person["husband"]]
```

Comment:
The code assumes that the 'wife' and 'husband' keys, if present, will always have valid names in the `Person.people` dictionary. This may not always be the case, and you should add error handling to manage cases where the spouse's name is not found.